python/ray/cloudpickle/cloudpickle.py

```diff
@@ -66,11 +65,6 @@
 except ImportError:
     _typing_extensions = Literal = Final = None
 
-if sys.version_info >= (3, 5, 3):
```

Comment:
Is the cloudpickle upgrade necessary for 3.11 support? If not, maybe we could separate these changes in another PR.

Reply:
I believe `//python/ray/tests:test_pydantic_serialization` or some similar test fails with the old version. That test still fails but with a different error now.

Reply:
Gotcha - commented on the pydantic dependency. 

Ideally, if this is not part of the python3.11 requirement, we should do this in another PR as well.

Reply:
Nvm - I think this needs to be part of the python3.11, I tested with the built wheel, and there's incompatibility with older cloud pickle source code.